### PR TITLE
Changing BucketID to be encoded in Base64

### PR DESCRIPTION
**Pull request by aarongalang**

### CODE REVIEW

File: pkg/azureutils/const.go [hunk 1]
```diff
@@ -36,6 +38,37 @@ const (
 	EnableLargeFileSharesField = "enablelargefileshares"
 )
 
+// bucketID is passed in DriverCreateBucket as a struct with subID and resource group for deletion
+type BucketID struct {
```
Comment (mukhoakash) [inline pkg/azureutils/const.go:42]: Lets create a separate pkg for "types" which will have the bucketID for now. We can use that pkg going ahead whenever we need to create objects for auth.

File: pkg/azureutils/cosi_api_ops.go [hunk 2]
```diff
@@ -97,8 +97,14 @@ func CreateBucket(ctx context.Context,
 func DeleteBucket(ctx context.Context,
 	bucketID string,
 	cloud *azure.Cloud) error {
+	//decode bucketID
+	id, err := decodeToBucketID(bucketID)
+	if err != nil {
+		return err
```
Comment (mukhoakash) [inline pkg/azureutils/cosi_api_ops.go:103]: Please add logs here before the return statement to debug easily

**Comment by mukhoakash** [comment]
@aarongalang Please add more logs in the newer changes. It becomes harder to debug otherwise.

File: pkg/azureutils/container_ops.go [hunk 3]
```diff
@@ -45,22 +45,37 @@ func createContainerBucket(
...
 	}
 	containerParams := make(map[string]string) //NOTE: Container parameters still need to be filled/implemented
-	return createAzureContainer(ctx, parameters.storageAccountName, key, bucketName, containerParams)
+
+	container, err := createAzureContainer(ctx, parameters.storageAccountName, key, bucketName, containerParams)
+	if err != nil {
+		return "", err
+	}
+	id := BucketID{
+		SubID:         cloud.SubscriptionID,
```
Comment (mukhoakash) [inline pkg/azureutils/container_ops.go:54]: If parameter is bringing in a subsId we need to use that. We have to put those checks in place. For now, we are not supporting a different subscription id to create bucket. But, we need to handle that here.

Comment (mukhoakash) [reply pkg/azureutils/container_ops.go:54]: If param.SubId is nil then we use cloud.subid

File: pkg/azureutils/const.go [hunk 4]
```diff
@@ -36,6 +38,37 @@ const (
 	EnableLargeFileSharesField = "enablelargefileshares"
 )
 
+// bucketID is passed in DriverCreateBucket as a struct with subID and resource group for deletion
```
Comment (mukhoakash) [inline pkg/azureutils/const.go:41]: Fix the comment to:
`// bucketID is returned by the DriverCreateBucket function call as an encoded string with the subID, resource group and the URL of the bucket. These details are required by DriverDeleteBucket and DriverGrantBucketAccess.`

File: pkg/azureutils/container_ops.go (hunk 3 above)
Comment (aarongalang) [reply pkg/azureutils/container_ops.go:55]: currently, we do not have a parameter for the subscriptionID in bucketclassparameters. I can add it if you would like.

**Review by mukhoakash — APPROVED** [review]
Waiting for a couple of small changes. Rest looks good

**Pull request merged**